tgui/packages/tgui/interfaces/common/sorting/SortableTable/index.tsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author mloccy (https://github.com/mloccy)
+ * @license ISC
+ */
+
+import { BooleanLike } from "common/react";
+import { Fragment } from "inferno";
+import { useLocalState, useSharedState } from "../../../../backend";
+import { Table, Tooltip } from "../../../../components";
+import { Header } from "../Header";
+import { SearchState, SortableTableHeaderConfig, SortableTableRowProps, SortState } from "./type";
+import { onSortClick, sortAndFilterRows } from "./utils";
+
+
+interface SortableTableHeaderRowProps {
+  config: SortableTableHeaderConfig[],
+  sortState: SortState,
+  setSortBy: (index: number) => void
+}
+
+const SortableTableHeaderRow = (props: SortableTableHeaderRowProps) => {
+  const { config, sortState, setSortBy } = props;
+
+  return (
+    <Table.Row header>
+      {config.map((config, index: number) => {
+        const { children, sortable, toolTipContent } = config;
+        const needsHeader = sortable === undefined ? true : sortable;
+
+        const sortDirection = !sortState
+          ? null
+          : index === sortState.index
+            ? sortState.dir
+            : null;
+        const sortByIndex = () => setSortBy(index);
+
+
+        if (toolTipContent) {
+          if (needsHeader) {
+
+            return (
+              <Tooltip content={toolTipContent}>
+                <Table.Cell header key={index}>
+                  <Header sortDirection={sortDirection} onSortClick={sortByIndex}>{children}</Header>
+                </Table.Cell>
+              </Tooltip>);
+          }
+          else {
+            return (
+              <Tooltip content={toolTipContent}>
+                <Table.Cell header key={index}>
+                  {children}
+                </Table.Cell>
+              </Tooltip>);
+          }
+        } else {
+
+          if (needsHeader) {
+            return (
+              <Table.Cell header key={index}>
+                <Header sortDirection={sortDirection} onSortClick={sortByIndex}>{children}</Header>
+              </Table.Cell>);
+          } else {
+            return (
+              <Table.Cell header key={index}>
+                {children}
+              </Table.Cell>);
+
+          }
+        }
+      })}
+    </Table.Row>
+  );
+};
+
+const SortableTableRow = (props: SortableTableRowProps) => {
+  const { cells, ...rest } = props;
```

Comment:
I had a bit of fun trying to follow `cells` around, would `columnConfig` be a more accurate name? (This is an actual question, not me being passive aggressive). If it seems like it would fit, consider using it. If not, could you explain what `cells` is here?